### ISSUE TITLE
The extension is broken since December

### ISSUE THREAD

**Issue by kingdonb** [weaveworks/vscode-gitops-tools#503]
### Expected behaviour

When 0.26.0 is used, it should not glitch out in the UI

### Actual behaviour

Super glitch out in the UI if you're running VSCode editor version from November or later. If you're running the editor version 1.84.2, the extension 0.26.0 behavior is perfect. If you are upgraded to the latest 1.85.2 then you're totally hosed.

### Steps to reproduce

See Sunkworks - Pt. 25 (Emerg. Maintenance) - youtube stream link (add later)

### Versions

We are going to rollback to 0.24.3 behavior, as soon as we confirm that version still works with the 1.85.2+ vscode release.

Version 0.27.0 will have the old (not-live updated UI) behavior where users need to manually refresh in the UI.
Version 0.28.0 will have the new (live updated UI) behavior where users never need to click refresh, with no UI glitching.

I've moved some milestones around, we may have prereleases between 0.27 and 0.28 that expect you're on the old 1.84 editor so you can get the new behavior - but until we determine how to fix it, if the issue can be fixed here, or if we need to fix it upstream, there will be no release that has the live updated UI feature. Sorry for releasing this prematurely!

Will update this issue when the rollback is complete, so users can decide which version they want (and so new users coming to the extension for the first time are not presented with a broken release, when they haven't opted into any prerelease build.)

**Comment by kingdonb** [comment]
The PR in #504 did not get the job done, #505 is going out as a second try to the "prerelease" edge branch now.

* #505 

I think this one worked. Git reset/checkout and revert with an edge branch in the middle is a bit harder than I remembered.

**Comment by kingdonb** [comment]
The rollback has been merged to main successfully. This means that our live update behavior is (temporarily) removed from the release.

In the mean time, if you're missing the live update function, you can pin VSCode to v1.84.2 and use v0.26.0 GitOps Tools.

(The v0.27.0 release which rolled back our live-update function is going out now, users should all be upgraded to it unless they are pinned to an older release/prerelease version.)

We will bring this feature back first in the prerelease channel, then as we vet it against different versions of VSCode, we may be able to promote it back to the main release again. We will not do any of that before v0.28.0 - patches in v0.27.0 will not bring back the live update behavior, but the prerelease channel might (should).

Hope this is clear for anyone that's still bearing with us using the extension! Sorry for the trouble, I *hate* asking people to pin themselves to a particular version, but we really can't avoid that since:

* #500 

That issue is crippling and makes the extension completely unusable for the vast majority of users. Many are already pinning.

**Comment by kingdonb** [comment]
The v0.27.0 release is out, the upcoming bump in edge may happen today or it may happen later.

Recommend don't use the edge/prerelease branch for now unless you're also pinning vscode to the October 2023 release or earlier! It may take some time to isolate the changes needed to fix this issue. We haven't found a way to contact the VSCode team directly about their changes yet, (probably could open an issue there now, but would rather go in better informed...)

**Comment by kingdonb** [comment]
There is a #flux-ecosystem channel on the CNCF slack that users can ask questions about the VSCode extension.

Today I acknowledged that development of the VSCode extension GitOps Tools has ceased, and there is no plan to work on v0.28.0 or future releases. This is my own opinion, anyone that is interested in further maintenance work for the VSCode extension is still welcome to talk to me about it. However, we did not budget time for this work given the shifting priorities.

It is not outside of the realm of possibilities that we would get new motivation to work on this. The redacted feature was really compelling, even if the private APIs got rugged by VSCode. I still wonder if there would be a way to implement this so it would not get broken, using public/ documented APIs instead of the ones that got rugged from under us.

**Comment by Poltergeisen** [comment]
I think weaveworks closed their doors. I'm not really expecting this addon to continue getting updates.

https://thenewstack.io/end-of-an-era-weaveworks-closes-shop-amid-cloud-native-turbulence/

**Comment by kingdonb** [comment]
@Poltergeisen That is correct, Weaveworks is closed, but that doesn't mean this extension cannot receive updates.

I have all the keys needed to do another release. If there is anyone else interested in working on it, I'd be more likely to push a release, as I don't really have the bandwidth to do both (release management + dev on the extension) - I'm only vaguely competent in typescript, and my coworker who did most of the development work is employed somewhere else now!

It might be better to fork, if there is interest, but it would certainly be less trouble for me to push another release out as Weaveworks (or 👻 🧟 undead ghost of weaveworks).

We made some questionable decisions leading up to the closing of Weaveworks, conference-driven development choices, and owing to the top-line issue I reported on here (Microsoft VSCode development changes to internal APIs we should not have used broke our hard work at the last minute) certain things about the current state of this extension are shall we say sub-optimal. I opened this issue to clarify how you can get the best experience, which is unfortunately not available on the latest version of VSCode.

Flux development is still full-steam ahead, many Flux maintainers are still engaged, the team is still growing, and the UI that I'm currently evaluating in opposition to this one is Headlamp. They have a Flux plugin which is in beta, not in the catalog yet (but it works) and that's the most active Flux UI project that I am aware of at this time.

The Headlamp UI experience for Flux in beta currently rivals Weave GitOps (OSS)

There are devs involved who have never been at Weaveworks, as well as some who were at Weaveworks before it closed.
